yt_dlp/extractor/nytimes.py

```diff
@@ -1,50 +1,42 @@
-import hmac
-import hashlib
-import base64
+import json
+import uuid
 
 from .common import InfoExtractor
 from ..utils import (
+    clean_html,
     determine_ext,
-    float_or_none,
+    extract_attributes,
+    get_elements_html_by_class,
     int_or_none,
-    js_to_json,
+    merge_dicts,
     mimetype2ext,
     parse_iso8601,
+    remove_end,
     remove_start,
+    str_or_none,
+    traverse_obj,
+    unified_strdate,
+    url_or_none,
 )
 
 
 class NYTimesBaseIE(InfoExtractor):
-    _SECRET = b'pX(2MbU2);4N{7J8)>YwKRJ+/pQ3JkiU2Q^V>mFYv6g6gYvt6v'
-
-    def _extract_video_from_id(self, video_id):
-        # Authorization generation algorithm is reverse engineered from `signer` in
-        # http://graphics8.nytimes.com/video/vhs/vhs-2.x.min.js
-        path = '/svc/video/api/v3/video/' + video_id
-        hm = hmac.new(self._SECRET, (path + ':vhs').encode(), hashlib.sha512).hexdigest()
-        video_data = self._download_json('http://www.nytimes.com' + path, video_id, 'Downloading video JSON', headers={
-            'Authorization': 'NYTV ' + base64.b64encode(hm.encode()).decode(),
-            'X-NYTV': 'vhs',
-        }, fatal=False)
-        if not video_data:
-            video_data = self._download_json(
-                'http://www.nytimes.com/svc/video/api/v2/video/' + video_id,
-                video_id, 'Downloading video JSON')
-
-        title = video_data['headline']
-
-        def get_file_size(file_size):
-            if isinstance(file_size, int):
-                return file_size
-            elif isinstance(file_size, dict):
-                return int(file_size.get('value', 0))
-            else:
-                return None
-
+    _GRAPHQL_API = 'https://samizdat-graphql.nytimes.com/graphql/v2'
+
+    @staticmethod
+    def _get_file_size(file_size):
+        if isinstance(file_size, int):
+            return file_size
+        elif isinstance(file_size, dict):
+            return int(file_size.get('value', 0))
+        else:
+            return None
+
+    def _extract_media_from_json(self, video_id, content_media_json):
         urls = []
         formats = []
         subtitles = {}
-        for video in video_data.get('renditions', []):
+        for video in content_media_json:
```

Comment:
IMO it makes more sense to do the traversal here instead of doing `.get('renditions')` etc 3+ times before passing it to this function
```suggestion
        for video in traverse_obj(content_media_json, ('renditions', ...)):
```